### DISCUSSION TITLE
New custom types enforcing a spec w/i the library?

### DISCUSSION THREAD

**Discussion by twcurrie** [pydantic/pydantic#2669] [Question] [answered]
I've used pydantic types like `AnyUrl` or `RedisDSN` which validate a string meets a specification, but I'm curious how open the author is to something like introducing additional types that validate the value meets a specification provided by NIST/IEEE/<insert other governing authority here>.  

I was specifically looking at NIST800-63b spec for passwords and wondering if it'd make sense to publish an outside library that validated per this spec, or offer a pull request to introduce that spec within the pydantic library.  However, I wonder to what direction the author (and/or community) wishes to go and to what level these specs would continue to be introduced into the library.

**Comment by samuelcolvin** [comment, marked as answer]
broadly speaking I'm fairly open to new types since they don't make the library much more complicated or effect performance.

However I've never heard of `NIST800-63b` and a short google doesn't look like it's particularly popular in the python world.

I would build your own type if you need it internally, and create a PR to add it to pydantic if sufficient people upvote this discussion or suggest they would use it.

**Reply by twcurrie** [reply]
Great, that's good to know. I'm not convinced that this is a good use of my time, but I thought the general question of new types was interesting enough to ask. Thanks for the prompt response and keep up the good work.
